### DISCUSSION TITLE
How to generate a set of replicable files and folders in the project?

### DISCUSSION THREAD

**Discussion by mrlucasrib** [copier-org/copier#663] [Q&A]
I would like to know if it is possible to create sub-templates and if so, how to use them. I have a project where we have a very repetitive job of creating new UseCases, where it is necessary to create a different folder with several files with very similar things between them. I wish there was a command like: `copier add usecase GetUserData` and then it would create some files in a folder with the same name.
Is this possible? if yes, how?

Also, someone in a specific part of the code (on a router) must add things to an already existing file. is this possible? How?

**Comment by wsascha** [comment]
@mrlucasrib, thanks for asking this! I have a similar problem.

If you really instantiate files from your template in different locations, I imagine what _could_ work is that you keep your templates in several subdirectories and instantiate them sequentially. But in my experience having a disjunct set of files is rarely the case.

My use case is more like that I have a "base" template and I want to add multiple optional extensions on top, ideally managed in separate projects.

**Comment by yajo** [comment]
You can have a template for use cases. Then:

    copier copy https://github.com/example/usecase-template.git usecase/GetUserData

**Reply by mrlucasrib** [reply]
There are many sub-templates. Do you know any other packages that have this functionality?

**Reply by wsascha** [reply]
Could you provide a bit more information on how the file & directory structure looks like? Will base- and sub-template modify the same files?

**Reply by mrlucasrib** [reply]
I have this project structure:
```
├── src
│   └── modules
│       ├── clients
│       │   ├── domain
│       │   ├── dto
│       │   ├── infra
│       │   ├── mappers
│       │   ├── repos
│       │   │   └── implementations
│       │   └── use_cases
│       │       ├── create_client
│       │       └── get_client
│       └── users
│           ├── domain
│           ├── dto
│           ├── exceptions
│           ├── infra
│           ├── mappers
│           ├── repos
│           │   └── implementations
│           └── use_cases
│               ├── confirm_user_account
│               │   └── dto
│               ├── create_user
│               ├── forgot_password
│               ├── login_user
│               ├── logout_user
│               ├── migrate_user
│               ├── resend_token
│               └── reset_password
│                   └── dto
```
I need in one command reply use_cases, like:
```
│   │       ├── infra
│   │       │   └── users_router.py
│            ├── create_user
│   │           │   ├── create_user_controller.py
│   │           │   └── create_user.py
```
where no `users_router.py` will be added at the end of the file.
And the others will be created according to the parameters passed to the CLI. Like the above command.
Also I want to make the structure of modules and repos replicable with a command.

**Comment by yajo** [comment]
You can use a subdirectory with a templated name.

**Reply by mrlucasrib** [reply]
It's work like a Ruby on Rails Generator? example:
`$ rails generate Controller MyControler`
or maybe Django 
`$ django-admin startapp blog`
